libethereum/ChainParams.cpp

```diff
@@ -37,6 +59,12 @@ ChainParams::ChainParams(string const& _json, h256 const& _stateRoot)
     *this = loadConfig(_json, _stateRoot);
 }
 
+ChainParams::ChainParams(std::string const& _s, AdditionalEIPs const& _additionalEIPs)
+  : ChainParams(_s)
+{
+    *this = addEIPs(_additionalEIPs);
```

Comment:
I think this approach is a little confusing (calling another ctor which sets some state and then overwriting the current instance) - I'd prefer if the addEIPs function either updates the current instances state (i.e. it's non-const) or we turn this ctor into a static function which calls `addEIPs ` and returns the result.

Reply:
Agree, this is complicated. I originally added only `addEIPs` and made it similar to `loadConfig` and `loadGenesis` (which create new instance out of the current one)

But now I think I should remove `addEIPs` and leave only this constructor.